NDcPP_v2_2e.adoc

```diff
@@ -2941,24 +2943,79 @@ _These requirements will be revisited as new TLS versions are standardized by th
 
-*FCS_TLSS_EXT.1.3* The TSF shall perform key establishment for TLS using [selection: _RSA with key size_ [selection: _2048 bits, 3072 bits, 4096 bits_]_, Diffie-Hellman parameters with size_ [selection: _2048 bits, 3072 bits, 4096 bits, 6144 bits, 8192 bits_]_, Diffie-Hellman groups_ [selection: _ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192, no other groups_]_, ECDHE curves_ [selection: _secp256r1, secp384r1, secp521r1] and no other curves_]].
+* _Diffie-Hellman parameters [selection: of size 2048 bits, of size 3072 bits, of size 4096 bits, of size 6144 bits, of size 8192 bits, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]_
+
+].
 
 *_Application Note {counter:appnote_count}_*
 
-The appropriate options shall be selected in the ST according to the key establishment options supported by the TOE. FMT_SMF.1 requires the configuration of the key agreement parameters to establish the security strength of the TLS connection.
+_The selected algorithms and key sizes must be consistent with FCS_CKM.2_
+
+_The following table provides the ST author with instructions for completing the selections._
 
-*FCS_TLSS_EXT.1.4* The TSF shall support [selection: _no session resumption or session tickets, session resumption based on session IDs according to RFC 4346 (TLS1.1) or RFC 5246 (TLS1.2), session resumption based on session tickets according to RFC 5077_].
+[cols=",,,",options="header",]
+|===
+| |RSA key establishment... |EC Diffie-Hellman key agreement... |Diffie-Hellman key agreement...
+|TLS 1.3 without TLS 1.2 or TLS 1.1 | Shall not select | Shall select at least one  | 
+|TLS 1.3 with TLS 1.2 or TLS 1.1 |See the following rows |Shall select at least one- see the following rows for additional instructions |
+|TLS 1.2 or TLS 1.1 with support for at least one TLS_RSA_WITH... ciphersuite |Shall select |N/A |N/A
+|TLS 1.2 or TLS 1.1 without support for any TLS_RSA_WITH... ciphersuite |Shall not select |N/A |N/A
+|TLS 1.2 or TLS 1.1 with support for at least one ECDHE ciphersuite |N/A |Shall select |N/A
+|TLS 1.2 or TLS 1.1 without support for TLS1.3 or any ECDHE ciphersuites |N/A |Shall not select |N/A
+|TLS 1.2 & TLS 1.1 with support for at least one DHE ciphersuite |N/A |N/A |Shall select
+|TLS 1.2 & TLS 1.1 without support for TLS 1.3 or any DHE ciphersuites |N/A |N/A |Shall not select
+|===
+
+_N/A indicates the options do not affect whether the selection is selected or not._
+
+_The RSA key size(s) specified in FCS_TLSS_EXT.1.3 are used for RSA key establishment._
+
+_Subsequent versions of this cPP will require support for the Supported Groups extension and prohibit use of legacy DH groups for key establishment._
+
+*FCS_TLSS_EXT.1.4* The TSF shall support [selection: _no session resumption, sessiion resumption based on session IDs according to RFC4346 (TLS1.1) or RFC5246 (TLS1.2), session resumption based on session tickets according to RFC5077 (TLS1.2), session resumption according to RFC8446 (TLS1.3)_].
```

Comment:
@qtrinh3  Check underline text here and in Appendix C. Underline is not showing up in GitHub, but we still need to add the underline text for pdf conversions. Syntax is [.underline]#underlined text here#. https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#ex-lines

Reply:
@qtrinh3 I agree and made those changes myself. Thanks!